### PR TITLE
[webview_flutter] iOS: Make `webViewWebContentProcessDidTerminate` invoke `onWebResourceError`

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

onWebResourceError has been handling `webContentProcessTerminated` error, however, `webViewWebContentProcessDidTerminate` callback on iOS is not handled. This PR fixes that. 
This will help apps to get a callback information when the webview content is not rendering; then those apps can choose to apply workarounds.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/61795

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/webview_flutter/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 1.0.6
+
+* iOS: `webViewWebContentProcessDidTerminate` is recognized in `WebView.onWebResourceError`.
```
Comment (amirh) [inline packages/webview_flutter/CHANGELOG.md:3]: nit: I found this description somewhat hard to understand, maybe something like: "Invoke the `WebView.onWebResourceError` on iOS when the webview content process crashes".

[assuming `webViewWebContentProcessDidTerminate` is only invoked when the process terminates due to some error, is this the case?]

**Review by amirh — APPROVED** [review]
LGTM modulo changelog nit

File: packages/webview_flutter/CHANGELOG.md (hunk 1 above)
Comment (cyanglaz) [reply packages/webview_flutter/CHANGELOG.md]: done!

[The documentation didn't specify what reasons the termination could be. I assume it is related to some sort of crashes. In the case I tried to fix, it was related to memory pressure]

**Comment by jbardu** [comment]
Sorry to add to this but I'm a bit lost with where this issue ended up. Was it simply integration of this PR to signal the app? webViewWebContentProcessDidTerminate can happen not just because of memory pressure but because the webview the app specifically wants to save, but is offscreen for a while .. is offscreen. In other words it will happen to everything but short lived, on-screen, web views.
I think auto-reloading web views terminated by IOS should be an explicit plugin boolean (default false) to encourage the app to handle it.

**Pull request merged**